tennis-scheduler-web-app/src/components/TennisCourtAddChange.jsx

```diff
@@ -0,0 +1,162 @@
+import { useState } from 'react'
+import { useEffect } from 'react';
+import { addTennisCourt, getTennisCourt, changeTennisCourt } from '../api/TennisCourtApi'
+import { useNavigate, useParams } from "react-router-dom";
+
+export const TennisCourtAddChange = () => {
+    const [address, setAddress] = useState({ country: "", city: "", street: "", number: "" })
+    const [surface, setSurface] = useState('')
+    const [image, setImage] = useState("")
+    const [tennisCourt, setTennisCourt] = useState({ name: "", surfaceType: "", description: "", image: "", address: address })
+    const [check, setCheck] = useState(false);
+    const [changedType, setChangedType] = useState('');
+    const [formErrors, setFormErrors] = useState({});
+    const [emptyForm, setEmptyForm] = useState(true);
+    const id = useParams().id
+    const navigate = useNavigate();
+
+    const handleChange = e => {
+        const { name, value } = e.target;
+        setTennisCourt((tennisCourt) => ({ ...tennisCourt, [name]: value }));
+    };
+
+    const handleChangeAddress = e => {
+        const { name, value } = e.target;
+        setAddress((address) => ({ ...address, [name]: value }));
+    };
+
+    const setImages = e => {
```

Comment:
Rename to onImageChange or handleImageChange maybe?
You already have a state named setState so it could be confusing to also have a method with same naming.